### ISSUE TITLE
@datadog/mobile-react-native-webview - Cant pass through ref to WebView

### ISSUE THREAD

**Issue by danbatsb** [DataDog/dd-sdk-reactnative#497]
**Describe what happened**

I'm trying to swap out the `react-native-webview` to use the DataDog `mobile-react-native-webview` but unfortunately I've come across an issue that breaks our existing views. We use the `injectJavaScript` method to run JS in the web view but this requires us to add a ref to the web view component. See the docs [here](https://github.com/react-native-webview/react-native-webview/blob/master/docs/Guide.md#the-injectjavascript-method) for an example of using this.

Currently with the DataDog webview I get the following: _Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?_ and our injected JS doesn't run.

**Steps to reproduce the issue:**

Add a ref to the web view and run some JS using the `injectJavaScript` method on the ref, as detailed in [this](https://github.com/react-native-webview/react-native-webview/blob/master/docs/Guide.md#the-injectjavascript-method) example.

**Describe what you expected:**
I can add a ref and the injected JS is run as it is when using the `react-native-webview`.

It looks like you might just need to wrap your WebView component inside `forwardRef` and then pass the `ref` into the `WebView` component.

Something like this seems to work
```ts
export const WebView = forwardRef((props: Props, ref: ForwardedRef<RNWebView | null>) => {
.....
return (
    <RNWebView
        ref={ref}
```

**Additional context**
```
"@datadog/mobile-react-native": "^1.7.1",
"@datadog/mobile-react-native-webview": "^1.7.1",
"@datadog/mobile-react-navigation": "^1.7.1"
```

**Comment by louiszawadzki** [comment]
Hi @danbatsb, thanks for reaching out!

This is indeed something we need to address, we'll tackle it very soon.
I'll let you know once this is released.

**Comment by louiszawadzki** [comment]
Hi @danbatsb,

The fix has been released under `1.8.0-rc1`.

You can use it as such:

```typescript
import React from 'react';
import WebView from '@datadog/mobile-react-native-webview';
import {WebView as RNWebView} from 'react-native-webview';

export const WebviewSite = () => {
  const ref = React.useRef<RNWebView>(null);

  return (
    <WebView
      source={{uri: 'https://my-site.com'}}
      allowedHosts={['my-site.com']}
      ref={ref}
    />
  );
};
```

Few caveats:
- the initial value in `useRef` should be `null`, not empty
- if you use typescript, the type of the ref has to be `WebView` from `react-native-webview` to ensure you the type matches your version of `react-native-webview`.

If that duplicated import is too much of a hassle for you let us know and we'll look into removing it :)

**Comment by danbatsb** [comment]
@louiszawadzki thats brilliant, thanks for getting this fixed so quickly.
